static/js/util.js

```diff
@@ -290,26 +290,41 @@ export function clean_user_content_links(html) {
             elt.removeAttribute("target");
         }
 
-        // Ensure that the title displays the real URL.
-        let title;
-        let legacy_title;
-        if (url.origin === window.location.origin && url.pathname.startsWith("/user_uploads/")) {
-            // We add the word "download" to make clear what will
-            // happen when clicking the file.  This is particularly
-            // important in the desktop app, where hovering a URL does
-            // not display the URL like it does in the web app.
-            title = legacy_title = $t(
-                {defaultMessage: "Download {filename}"},
-                {filename: url.pathname.slice(url.pathname.lastIndexOf("/") + 1)},
-            );
+        const is_inline_image =
```

Comment:
I'm still not a huge fan of this conditional. It's the only part of the function that has to look outside of the passed in element to make a decision. This is due to how the markdown processor constructs a wrapper div for inline elements: 

https://github.com/zulip/zulip/blob/af5d0d6f5e5444332f9f8e565d97f4acdceaa72f/zerver/lib/markdown/__init__.py#L652

It might be a bit cleaner if the Markdown processor itself added some class to the <a> to distinguish it as an inline image, or perhaps even adding the `aria-label` attribute itself.